src/amo/components/SearchSort/SearchSort.scss

```diff
@@ -65,6 +73,10 @@
     background: transparentize($link-color, 0.8);
     box-shadow: none;
   }
+
+  @include respond-to(large) {
```

Comment:
Yep, I completely agree about the trade-offs that's why I mentioned the header approach is not equal to a viewport size based decision. Especially since the header available has no granularity say compared to using a device database to customize responses to the device more closely.

One place where with-holding content might be a benefit in the future is if significant page weight could be saved on mobile as opposed to just hiding the content via styling. 

This is something that can be considered on a case-by-case and as with any optimizations we can look at it later when we're specifically looking at ways to reduce page size / increase perf.